scripts/pkg/build_templates/opensearch/deb/debian/preinst

```diff
@@ -23,6 +25,25 @@ if command -v systemctl >/dev/null && systemctl is-active opensearch-performance
     systemctl --no-reload stop opensearch-performance-analyzer.service
 fi
 
+# Check if OPENSEARCH_INITIAL_ADMIN_PASSWORD is defined
+# TODO:
+# 1. This check will need to be modified if there will be a min dist for deb in future (currently there is none)
+# 2. Currently, the demo config setup is defined to run, in postinst, if `opensearch-security` is present. Cannot apply the same check here since the plugins folder is not available yet.
+
+# Check if this is an upgrade by checking whether opensearch already exists
+OPENSEARCH_INSTALLED=$(dpkg -l | grep opensearch)
+
+OPENSEARCH_REQUIRED_VERSION="2.12.0"
+COMPARE_VERSION=`echo $OPENSEARCH_REQUIRED_VERSION $OPENSEARCH_VERSION | tr ' ' '\n' | sort -V | uniq | head -n 1`
```

Comment:
This assign the "lowest version", the name `COMPARE_VERSION` does not make that obvious.  I wonder if renaming this variable could help figure out more easily what the script does.

An alternative option may be to extract the logic in a function:

```sh
lowest_version() {
  echo "$@" | tr ' ' '\n' | sort -V | uniq | head -n 1
}

COMPARE_VERSION=`lowest_version $OPENSEARCH_REQUIRED_VERSION $OPENSEARCH_VERSION`
```